File: sentimental-credit/credit.py
# TODO
from cs50 import get_int
from cs50 import get_string

credit_card_number = get_int("Credit Card Number: ")
credit_card_length =  len(str(credit_card_number))

#get every number
number = [int(a) for a in str(credit_card_number)]
#Pass number
original_length = [int(a) for a in str(credit_card_number)]
#Multiply every 2nd number and get sum if greater than 10
z = credit_card_length - 1
while z > 0:
    number[z-1] = number[z-1] * 2
    if number[z-1] > 9:
        double_digit = [int(a) for a in str(number[z-1])]
        number[z-1] = double_digit[0] + double_digit[1]
    z = z - 2

#Get credit card number value
credit_card_sum = sum(number)

#Classify Credit card
#if credit card starts with 4 and length is 13 or 16
if (((original_length[0] == 4 and credit_card_length == 13) or (original_length[0] == 4 and credit_card_length == 16)) and credit_card_sum  % 10 == 0):
    print("VISA")
#if credit card starts with 34, 37 and length is 15
elif ((original_length[0] == 3 and (original_length[1] == 4 or original_length[1] == 7)) and credit_card_length == 15 and credit_card_sum % 10 == 0):
    print("AMEX")
#if credit card starts with 51, 52, 53, 54, 55 and length is 16
elif ((original_length[0] == 5 and (original_length[1] == 1 or original_length[1] == 2 or original_length[1] == 3 or original_length[1] == 4 or original_length[1] == 5)) and credit_card_sum % 10 == 0 and credit_card_length == 16):
    print("MASTERCARD")
else:
    print("INVALID")

File: credit/credit.c
#include <cs50.h>
#include <string.h>
#include <stdio.h>
#include <stdlib.h>

long credit_card_number;
int credit_card_length;

int main(void)
{
    do
    {
        credit_card_number = get_long("What's your credit card number? ");
        long credit_card_number_container = credit_card_number;


        while (credit_card_number_container > 0) {
            credit_card_number_container = credit_card_number_container / 10;
            credit_card_length++;
        }

        int number[credit_card_length], originallength[credit_card_length];

        for (int i = 0; i < credit_card_length; i++)
        {
            number[i] = credit_card_number % 10;
            credit_card_number = credit_card_number / 10;
        }

        for (int i = 1; i < credit_card_length; i++) {
            originallength[i] = number[i];
        }

        for (int i = 1; i < credit_card_length; i+=2) {
            number[i] = number[i] * 2;
        }


        int countsum = 0;
        int temp;

        for (int i = 0; i < credit_card_length; i++)
        {
            temp = (number[i] % 10) + (number[i]/10 % 10);
            countsum = countsum + temp;
        }

        if (((originallength[12] == 4 && credit_card_length == 13) || (originallength[15] == 4 && credit_card_length == 16)) && countsum  % 10 == 0 )
        {
            printf("VISA\n");
        }
        else if (originallength[14] == 3 && (originallength[13] == 4 || originallength[13] == 7) && countsum  % 10 == 0  && credit_card_length == 15)
        {
            printf("AMEX\n");
        }
        else if (originallength[15] == 5 && (originallength[14] == 1 || originallength[13] == 2 || originallength[13] == 3 || originallength[13] == 4 || originallength[13] == 5)&& countsum  % 10 == 0 &&  credit_card_length == 16)
        {
            printf("MASTERCARD\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }

    while (credit_card_number < 0);
    
}